src/main/java/com/github/osisoft/ocs_sample_library_preview/dataviews/DataViewClient.java

```diff
@@ -108,12 +109,17 @@ private String getRequestResponse(URL url, String method, String body) throws Sd
         return response;
     }
 
-    private ResponseWithLinks getRequestResponseWithLinks(URL url, String method, String body) throws SdsError {
+    private ResponseWithLinks getRequestResponseWithLinks(URL url, String method, String body, Map<String, String> headers) throws SdsError {
         HttpURLConnection urlConnection = null;
         ResponseWithLinks response = new ResponseWithLinks();
 
         try {
-            urlConnection = baseClient.getConnection(url, method);
+            if (headers == null) {
```

Comment:
Would it be possible to handle this null check in base client instead? That way we would avoid having to null check every time we call getConnection with headers.

Reply:
I believe that a function overload as opposed to passing null would follow the pattern of baseClient.getConnection better. This would be additional code, however.

Reply:
I think it would be easier if I just made a new commit with all of this to show it better, but I think the way it is now might end up being better. It's a lot cleaner, just has a null check that, as yall correctly point out, is not very idiomatic

Reply:
actually, i realized that even that won't work because if a value of true is provided to the verbosity parameter, it results in a null headers object being passed. So ultimately, I just have every data overload funneling into the one that accepts all parameters and verbosity, then that overload checks for verbosity. If it's false, it makes the headers object and returns the overload that accepts a headers object; if it's true, then it returns the overload that does not accept the headers object (and creates one with the base client later). 

So this approach has two return statements in each of the final data retrieval overloads for stored and interpolated, but there's no null checks anymore. The URL Connection handling was also split across three functions to eliminate code duplication across the headers/no headers approach, each creating their URL Connection, then passing to a third overload that accepts a URL connection.

Reply:
this feels...organic (plus it actually works :) )